### PR TITLE
Per fetch file token for blob url

### PR DESCRIPTION
**Pull request by gterzian**
Depends on https://github.com/servo/servo/pull/25724

First step of https://github.com/servo/servo/issues/25226

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by gterzian** [comment]
```
  ▶ Unexpected subtest result in /FileAPI/url/cross-global-revoke.sub.html:
  │ FAIL [expected PASS] It is possible to revoke same-origin blob URLs from a different worker global.
  └   → assert_unreached: Should have rejected: undefined Reached unreachable code
  ▶ Unexpected subtest result in /FileAPI/url/url-with-xhr.any.worker.html:
  │ FAIL [expected PASS] XHR of a revoked URL should fail
  │   → assert_equals: expected 0 but got 416
  │ FAIL [expected PASS] Appending a query string should cause XHR to fail
  │   → assert_equals: expected 0 but got 416
  │ FAIL [expected PASS] Appending a path should cause XHR to fail
  │   → assert_equals: expected 0 but got 416
  │ 
  │ xhr_should_fail/result2</xhr.onreadystatechange<@http://web-platform.test:8000/FileAPI/url/url-with-xhr.any.js:27:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:2024:25
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:2049:35
  ▶ Unexpected subtest result in /FileAPI/url/url-with-xhr.any.html:
  │ FAIL [expected PASS] XHR of a revoked URL should fail
  │   → assert_equals: expected 0 but got 416
  │ FAIL [expected PASS] Appending a query string should cause XHR to fail
  │   → assert_equals: expected 0 but got 416
  │ FAIL [expected PASS] Appending a path should cause XHR to fail
  │   → assert_equals: expected 0 but got 416
  │ 
  │ xhr_should_fail/result2</xhr.onreadystatechange<@http://web-platform.test:8000/FileAPI/url/url-with-xhr.any.js:27:7
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:2024:25
  └ Test.prototype.step_func/<@http://web-platform.test:8000/resources/testharness.js:2049:35
```

**Comment by gterzian** [comment]
Ok this one seems good to go(still depends on first merging https://github.com/servo/servo/pull/25724)

@Manishearth r?

**Comment by gterzian** [comment]
Rebased and ready for review. 

@Manishearth r?

**Comment by gterzian** [comment]
@Manishearth r?

**Comment by Manishearth** [comment]
Yeah, I have this saved, hope to get around to it today

**Review by Manishearth — APPROVED** [review]
nice design!

File: components/net_traits/filemanager_thread.rs [hunk 1]
```diff
@@ -16,6 +16,17 @@ use uuid::Uuid;
...
 
+/// A token modulating access to a file for a blob URL.
+pub enum FileTokenCheck {
+    /// Checking against a token not required,
+    /// used for accessing a file
+    /// that isn't linked to from a blob URL.
+    NotRequired,
+    /// Checking against token required,
+    /// the None case always fails.
+    Required(Option<Uuid>),
```
[the remark this answers is not in the document]
Comment (gterzian) [reply components/net_traits/filemanager_thread.rs]: Yes that's a good idea.

**Comment by Manishearth** [comment]
@bors-servo r+

thanks!

**Pull request merged**